HardwareCheckout/timer.py

```diff
@@ -0,0 +1,38 @@
+import os
+from base64 import urlsafe_b64encode
+
+from flask import Blueprint
+
+
+def Timer(path):
+    timer = Blueprint('timer', __name__)
+    map = {}
+
+    def add_timer(callback, timeout):
+        timeout = int(timeout)
+        key = urlsafe_b64encode(os.urandom(32)).decode('latin-1').rstrip('=')
+        if key in map:
+            raise Exception('random number collision (!)')
+        map[key] = callback
+        if os.system('systemd-run --on-active=%i curl -X POST http://localhost:5000%s/%s' % (timeout, path, key)):
```

Comment:
Yeah, `os.system` is not ideal here, it's just the quickest thing I could get to work. I'd definitely be interested in looking at `threading.Timer`. Whatever we do, we need to have a flask context so that sqlalchemy and socketio work from within the callback. I figured the easiest way to do that would be to access randomly generated URLs, but there might be something else available, we'd just need to figure out what it is.

Reply:
Cool. I'll open an issue for that then.